functors task (1)/student/include/impl.h

```diff
@@ -1,48 +1,73 @@
 #pragma once
-#include "bar_serving.h"
 #include <functional>
+#include "bar_serving.h"
 
 /**
  * @todo Implement functor-generator that will return next beer brand (cyclic)
- * @example HoeGaarden -> Corona -> Carlsberg -> Bud -> ZlataPraha -> Leffe -> HoeGaarden...
+ * @example HoeGaarden -> Corona -> Carlsberg -> Bud -> ZlataPraha -> Leffe ->
+ * HoeGaarden...
  */
-struct BeerOrganizer
-{
+struct BeerOrganizer {
+ public:
+  BeerBrand operator()() {
+    BeerBrand bb = static_cast<BeerBrand>(index);
+    ++index;
+    if (index == 7) {
+      index = 1;
+    }
+    return bb;
+  }
+
+ private:
+  int index = 1;
 };
 
 /**
- * @todo Implement unary predicate function that will determine if beer is expensive
+ * @todo Implement unary predicate function that will determine if beer is
+ * expensive
  * @return true in case if beer is expensive, false if not
  *
  * @note Only Corona and HoeGaarden are expensive
  */
-bool isExpensiveBeer(/**???*/)
-{
+bool isExpensiveBeer(BeerBrand const& bb) {
+  if (BeerBrand::Corona == bb || BeerBrand::HoeGaarden == bb) {
```

Comment:
why not
return beer == BeerBrand::Corona || beer == BeerBrand::HoeGaarden;

avoid explicit true/false in your code!